UnicornEngine/video/include/unicorn/video/vulkan/VkMesh.hpp

```diff
@@ -55,47 +56,75 @@ class VkMesh
      * @brief Allocation on GPU
      */
     void AllocateOnGPU();
+
     /**
      * @brief Deallocation on GPU
      */
     void DeallocateOnGPU();
+
     /**
      * @brief Returns vertex buffer
      * @return vulkan buffer
      */
-    vk::Buffer GetVertexBuffer();
+    vk::Buffer GetVertexBuffer() const;
+
     /**
      * @brief Returns index buffer
      * @return vulkan buffer
      */
-    vk::Buffer GetIndexBuffer();
+    vk::Buffer GetIndexBuffer() const;
     /**
      * @brief Matrix of transformations
      * @return model matrix
      */
-    const glm::mat4& GetModel() const;
+    const glm::mat4& GetModelMatrix() const;
+
     /**
      * @brief Returns vertices size
      * @return size of vertices
      */
-    uint32_t VerticesSize();
+    uint32_t VerticesSize() const;
+
     /**
     * @briefReturns indices size
     * @return size of indices
     */
-    uint32_t IndicesSize();
+    uint32_t IndicesSize() const;
+
+    bool IsColored() const;
+
+    bool IsWired() const;
+
+    glm::vec3 GetColor() const;
+
+    /*
+     * @brief Material in vulkan is combination of descriptor set and binded data
+     * //TODO
+     */
+    void SetMaterialHandle(uint32_t handle);
+
+    void OnMaterialUpdated(Mesh* mesh);
+
+    uint32_t GetMaterialHandle() const;
     /**
      * @brief Signal for command buffer reallocation
      */
     wink::signal<wink::slot<void(VkMesh*)>> ReallocatedOnGpu;
+    wink::signal<wink::slot<void(Mesh*, VkMesh*)>> MaterialUpdated;
 private:
     bool m_valid;
+    bool m_isColored;
+    bool m_isWired;
+
     vk::Device m_device;
     vk::PhysicalDevice m_physicalDevice;
     vulkan::Buffer m_vertexBuffer, m_indexBuffer;
-    geometry::Mesh& m_mesh;
     vk::CommandPool m_pool;
     vk::Queue m_queue;
+
+    Mesh& m_mesh;
```

Comment:
Instances of this won't be copyable, is it okay?

Reply:
VkMesh never copy to another VkMesh.

Reply:
@Warezovvv Maybe explicitly delete this functional then?